lib/elixir/lib/macro.ex

```diff
@@ -2674,6 +2674,69 @@ defmodule Macro do
     end
   end
 
+  defp dbg_ast_to_debuggable({:with, meta, args} = ast, _env) do
+    {opts, clauses} =
+      List.pop_at(args, -1)
+
+    modified_clauses =
+      Enum.flat_map(clauses, fn
+        {:<-, _meta, [left, right]} = clause ->
+          quote do
+            [
+              line = Keyword.get(unquote(meta), :line),
+              code = unquote(Macro.escape(clause)),
+              value = unquote(right),
+              unquote(:<-)({{line, code, value}, unquote(left)}, {{line, code, value}, value})
+            ]
+          end
+
+        # Other expressions are omitted.
+        expr ->
+          [expr]
+      end)
+
+    modified_opts =
+      case opts do
+        [do: do_code] ->
+          else_code =
+            quote do
+              [
+                unquote(:->)(
+                  [{unquote({:unmatched_ast, [], Elixir}), unquote({:result, [], Elixir})}],
```

Comment:
FYI: You typically want to use [`unique_var`](https://hexdocs.pm/elixir/1.17.3/Macro.html#unique_var/2) for this, there are other examples in the dbg code around.